### PR TITLE
feat: charmil init command

### LINKED ISSUE #164: Create go templates for starter CLI
This will include 
- shell commands - cmd/CLI_NAME, pkg/cmd/root, etc
- License 
- Contributing.md
- .gitignore
- .golangci.yaml
- .goreleaser.yaml
- Makefile
- Readme.md

.github, .githooks etc can be specific to repository, hence should not be our business.

### PR DESCRIPTION
**Pull request by ankithans**
Closes #164 #165 

### Tasks
Charmil init command will clone the starter repo and set it up according to details provided by the user.
- [x] Get prompts for Owner, repo and CLI Name
- [x] Configure the [starter repo](https://github.com/ankithans/charmil-starter-template) with templates
- [x] Clone the starter repo
- [x] Replace template placeholders with prompts taken from user - ignore .git, .github, .chglog etc

### Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation change

### CODE REVIEW

**Comment by ankithans** [comment]
Working: 

image

**Comment by ankithans** [comment]
@wtrocki can I transfer this repo https://github.com/ankithans/charmil-starter-template to aerogear?

File: pkg/cmd/init/init.go [hunk 1]
```diff
@@ -13,8 +37,155 @@ func InitCommand(f *factory.Factory) *cobra.Command {
...
+	}
+
+	err := filepath.Walk(path,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+
+			// ignore folders
+			ignoreSlice := []string{".git", ".github", ".chglog", ".goreleaser.yml", "CONTRIBUTING.md", "bin"}
```
Comment (wtrocki) [inline pkg/cmd/init/init.go:138]: Why we have this?

**Review by wtrocki — APPROVED** [review]

File: pkg/cmd/init/init.go (hunk 1 above)
Comment (ankithans) [reply pkg/cmd/init/init.go:138]: to ignore these folders, while applying template

Comment (wtrocki) [reply pkg/cmd/init/init.go:138]: Yes. I get it but why? Those are useful files that are part of starter

Comment (ankithans) [reply pkg/cmd/init/init.go:138]: I didn't have any templates defined in these folders and some of these were creating some error while applying templates. So I had to do this.
But these will be generated untouched as it is.

**Comment by namit-chandwani** [comment]
@ankithans I'm getting a `repository already exists` error every time I try to use `charmil init`:
```bash
./charmil init

GitHub Organization or Username: namit-chandwani
GitHub Repo Name: trial-cli
CLI Name: trial
repository already exists
```

Can you please look into this?

**Comment by ankithans** [comment]
okay got it, this works if .git doesn't exist. Will fix it.

**Pull request merged**